components/tooltip/tooltip-help.js

```diff
@@ -0,0 +1,92 @@
+import '../colors/colors.js';
+import '../tooltip/tooltip.js';
+import { css, html, LitElement } from 'lit';
+import { bodySmallStyles } from '../typography/styles.js';
+import { FocusMixin } from '../../mixins/focus-mixin.js';
+import { FocusVisiblePolyfillMixin } from '../../mixins/focus-visible-polyfill-mixin.js';
+
+/**
+ * A component used to display additional information when users focus or hover on certain text.
+ * @slot - Default content placed inside of the tooltip
+ * @fires d2l-tooltip-show - Dispatched when the tooltip is opened
+ * @fires d2l-tooltip-hide - Dispatched when the tooltip is closed
+ */
+class HelpTooltip extends FocusMixin(FocusVisiblePolyfillMixin(LitElement)) {
+
+	static get properties() {
+		return {
+			/**
+			 * REQUIRED: The text that will be rendered as the Help Tooltip component.
+			 * @type {String}
+			 */
+			text: { type: String }
+		};
+	}
+
+	static get styles() {
+		return [bodySmallStyles, css`
+			:host {
+				display: inline-block;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			#d2l-tooltip-help-text {
+				text-decoration-color: var(--d2l-color-galena);
+				text-decoration-line: underline;
+				text-decoration-style: dashed;
+				text-decoration-thickness: 1px;
+				text-underline-offset: 0.15rem;
+			}
+			#d2l-tooltip-help-text:focus {
+				outline-style: none;
+			}
+			#d2l-tooltip-help-text.focus-visible {
+				border-radius: 0.025rem;
+				outline: 2px solid var(--d2l-color-celestine);
+				outline-offset: 0.15rem;
+				text-underline-offset: 0.05rem;
+			}
+		`];
+	}
+
+	static get focusElementSelector() {
+		return 'span';
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+		if (!this.text || this.text.length === 0) {
+			console.warn('Help Tooltip component requires text.');
+		}
+	}
+
+	render() {
+		return html`
+			<span 
```

Comment:
Could we just use a `<button>` here instead and style it to look how we want? Then we wouldn't need to listen for key down or put a `tabindex` on it.

Reply:
Yeah, we could. I think this is something we kept going back and forth on with @dbatiste and @geurts.

There have been several comparisons done for things like how the screen reader would read the component depending on whether it's a `span` or a `button`.

We last settled on [going with a `span`](https://d2l.slack.com/archives/C03H8KH2FKN/p1653919910721479?thread_ts=1653676341.277049&cid=C03H8KH2FKN), @geurts do you remember why we preferred that over just using a `button` tag?

Reply:
Yeah I'd be curious about the reasoning. In theory, `<span role="button">` and `<button>` should behave exactly the same, which is why the general advice is to just always use the correct native element for the job instead of adapting another one using ARIA. Plus `<button>` gives you built-in goodness like handling click/keyboard and tab stops.

Reply:
After [discussing with Jeff and Dave B](https://d2l.slack.com/archives/C03H8KH2FKN/p1654633432212059?thread_ts=1654613644.917059&cid=C03H8KH2FKN), we agree that using a `<button>` works better here.